### DISCUSSION TITLE
How to test select createOptionUsing

### DISCUSSION THREAD

**Discussion by AshrafBasry** [filamentphp/filament#10381] [Help] [answered]
I searched the docs for how to test select createOptionUsing modal but i couldn't find any
```
Forms\Components\Select::make('owner_id')
    ->relationship('owner', 'name')
    ->searchable()
    ->preload()
    ->createOptionForm([
        Forms\Components\TextInput::make('name')
            ->required()
            ->maxLength(255),
    ])
   ->createOptionUsing(function ($data) {
          Owner::create($data);
    })
    ->required()
```
how to test if this option form works ?

**Comment by AshrafBasry** [comment]
this is what i am trying to do right now
```
Livewire::test(BookResource\Pages\CreateBook::class)
            ->assertActionExists('createOption')
            ->mountAction('createOption')
            ->setActionData([
                'name' => 'Test Owner',
            ])
            ->callMountedAction()
            ->assertHasNoActionErrors();
```

but i can't get the select create action name so i used then action name from
\Filament\Forms\Components\Select
```
public function getCreateOptionActionName(): string
{
    return 'createOption';
}
```

**Reply by leandrocfe** [reply, marked as answer]
```php
Livewire::test(BookResource\Pages\CreateBook::class)
->call('mountFormComponentAction', 'data.owner_id', 'createOption')
->set('mountedFormComponentActionsData.0.name', 'some value')
->call('callMountedFormComponentAction')
->assertHasNoFormErrors();
```

**Reply by AshrafBasry** [reply]
Thanks :handshake: it worked.
where can i find more docs about `mountFormComponentAction`

**Reply by leandrocfe** [reply]
There isn't I think, but you can check the source code

**Reply by anindyoy** [reply]
> ```
> Livewire::test(BookResource\Pages\CreateBook::class)
> ->call('mountFormComponentAction', 'data.owner_id', 'createOption')
> ->set('mountedFormComponentActionsData.0.name', 'some value')
> ->call('callMountedFormComponentAction')
> ->assertHasNoFormErrors();
> ```

I've tried these in filament v4, and I got error 

  Unable to call component method. Public method [mountFormComponentAction] not found on component
  
any update how to test it?
